hub/src/main/java/com/hub/HubController.java

```diff
@@ -0,0 +1,117 @@
+package com.hub;
+
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import java.io.OutputStream;
+
+import com.google.gson.Gson;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+
+import javax.annotation.PostConstruct;
+import java.net.HttpURLConnection;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.nio.charset.StandardCharsets;
+import java.util.List;
+
+@RestController
+public class HubController {
+
+    private final ServerHub hub;
+    private final HubRepository repository;
+
+    public HubController(HubRepository repository){
+        this.repository = repository;
+        this.hub = new ServerHub();
+    }
+
+    @PostConstruct
+    private void mapSetUp(){
+        List<HubEntry> servers = repository.findAll();
+        for (HubEntry entry : servers) {
+            this.hub.addServer(entry.getId(), entry.getServerAddress());
+        }
+    }
+
+    @PostMapping("/hub")
+    protected Long addServer(@RequestBody String json) throws Exception {
+        String address = parseAddressFromJsonString(json);
+
+        //TODO: למעשה, do we really need to set the server address twice and save twice?
```

Comment:
short answer - no. Why are you doing that now?

Reply:
It's required to generate an ID the first time around and to put that generatedValue in the address the second time around

Reply:
ah, indeed. why set it the first time then? why not just save, get the id, and then set the server address?